core/tests/conftest.py

```diff
@@ -1,6 +1,35 @@
+from pathlib import Path
+
 import pytest
 from typing import Callable
+import subprocess
 from testcontainers.core.container import DockerClient
+import sys
+
+PROJECT_DIR = Path(__file__).parent.parent.parent.resolve()
+
+
+def pytest_configure(config: pytest.Config) -> None:
+    """
+    Add configuration for custom pytest markers.
+    """
+    config.addinivalue_line(
+        "markers",
+        "inside_docker_check: test used to validate DinD/DooD are working as expected",
+    )
```

Comment:
is this just for filtering in/out dind/dod tests?

Reply:
That is to filter out tests that run within the `test_dind` and `test_dood` tests.
Inside this tests now the docker container is build and executed in a `dind` and `dood` environment.
This way we have proper test that `dind` and `dood` actually works.

I left the old `make test-dind`  (which is not correct because it's acutall DooD) untouched.
It would actually be better to remove it not only because of the wrong name.
It also doesn't succeed because some things as some *auth* tests seem to have troubles with TLS settings. 
I don't think the complete test suit has to pass inside a dind/dood env.
In the end these test are integration tests and should be treated as such.

Reply:
lets remove the things that don't make sense

Reply:
perhaps in a separate PR maybe to make the history clearer